OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -322,6 +216,174 @@ void TracerThread::Run(
     close(fd);
   }
   fds_to_ring_buffer.clear();
+
+  for (int uret_probe_fd : uret_probes_fds) {
+    perf_event_disable(uret_probe_fd);
+    close(uret_probe_fd);
+  }
+}
+
+void TracerThread::ProcessContextSwitchEvent(const perf_event_header& header,
+                                             PerfEventRingBuffer& ring_buffer) {
+  ContextSwitchPerfEvent event;
+  ring_buffer.ConsumeRecord(header, &event.ring_buffer_record);
+  ++sched_switch_count_;
+  pid_t tid = event.GetTid();
+  uint16_t cpu = static_cast<uint16_t>(event.GetCpu());
+  uint64_t time = event.GetTimestamp();
+
+  if (event.IsSwitchOut()) {
+    listener_->OnContextSwitchOut(ContextSwitchOut(tid, cpu, time));
```

Comment:
I know that as of now it creates problems with identifying the type of event in the other thread, and I know that at the moment OnContextSwitchIn/Out don't take long, but should we call this listener also in the processing thread for consistency?